corehq/apps/userreports/sql/data_source.py

```diff
@@ -58,7 +58,6 @@ def _db_columns(self):
         # This explicitly only includes columns that resolve to database queries.
         return [c for c in self.inner_columns if not isinstance(c, CalculatedColumn)]
 
-    @memoized
```

Comment:
I'm curious, what was the motivation for removing memoization?

Reply:
Scenario for which this change is done - User's role has permission 'Access to all the locations' enabled and report is saved at that time. Afterwards, permission is removed and now user is assigned specific location(s). Reports sent to recipients still contain rows from locations that user  doesn't have access now. This data, that is sent in email,  is currently 'memoized' and an actual call to DB is not made with the updated location filters. This is why it was removed at places to allow DB call.

Reply:
@akashkj I understand the scenario you are highlighting but its not clear why it would happen because of the memoization here.
Is this in case the report is being sent out to multiple users and the data is fetched only once and then re-used for all users?

Reply:
@akashkj Please share link to the code where the filtering is happening.

Reply:
https://github.com/dimagi/commcare-hq/blob/akj/indiv-197-dfci-custom-update-rule/corehq/apps/reports/sqlreport.py#L288

This uses filters that are applied on the report. It is called from `get_data()` from which `memoized` is removed here.

Reply:
The code has been reworked a bit since the `@memoized` decorators were removed, pushing the location safety down into the filters themselves.  Is it possible the memoization can be added back now?